### PR TITLE
Adding Launcher application to Display Cluster

### PR DESCRIPTION
**Pull request by bilgili**
Dont merge or review yet. It is to show how deflect stream is being used.

### CODE REVIEW

File: apps/Launcher/FileBrowser.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        rootObject->setProperty( rootFolderOption.toStdString().c_str(),
+                                 parser.value( rootFolderOption ));
+
+        filters.append( "dcx" );
+
+        rootObject->setProperty( filterProperty.toStdString().c_str(),
+                                 filters );
+
+        QObject::connect( rootObject, SIGNAL(itemSelected(QString)),
+                          &fileBrowser, SLOT(_onItemSelected(QString)));
```
Comment (rdumusc) [inline apps/Launcher/FileBrowser.cpp]: Just one note: the content of onItemSelected(QString) can be moved to a lambda function here, this removes the rather long forwarding and the need for a private slot in the header

File: apps/Launcher/FileBrowser.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
+/*********************************************************************/
+/* Copyright (c) 2013, EPFL/Blue Brain Project                       */
```
Comment (tribal-tec) [inline apps/Launcher/FileBrowser.cpp]: 2016?

File: apps/Launcher/qml/FileBrowser/FileBrowser.qml [hunk 3]
```diff
@@ -0,0 +1,241 @@
+/****************************************************************************
+**
+** Copyright (C) 2014 Digia Plc and/or its subsidiary(-ies).
```
Comment (tribal-tec) [inline apps/Launcher/qml/FileBrowser/FileBrowser.qml]: it this copyright valid or should be 'ours'?

**Comment by tribal-tec** [comment]
Do we need license/acknowledgments for the icons?

File: apps/CMakeLists.txt [hunk 4]
```diff
@@ -5,5 +5,6 @@
 include_directories(${PROJECT_SOURCE_DIR}/dc/core)
 
 add_subdirectory(DisplayCluster)
+add_subdirectory(Launcher)
```
Comment (tribal-tec) [inline apps/CMakeLists.txt]: add conditional if(TARGET DeflectQt)

File: apps/Launcher/ImageProvider.h [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#include <QtQuick/QQuickImageProvider>
+
+class ImageProvider : public QQuickImageProvider
+{
+public:
+    ImageProvider( const QSize defaultSize = QSize( 512, 512 ));
+
+    QImage requestImage( const QString& filename,
+                         QSize* size,
+                         const QSize& requestedSize );
```
Comment (tribal-tec) [inline apps/Launcher/ImageProvider.h]: final

File: apps/Launcher/Launcher.cpp [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+const QUrl qmlFile( "qrc:/qml/qml/main.qml" );
+
+struct Launcher::Impl
+{
+    Impl( Launcher& launcher )
+        : launcher_( launcher )
+    {
+
+        QGuiApplication::setApplicationVersion(
+                    QString::fromStdString( deflect::Version::getString( )));
```
Comment (tribal-tec) [inline apps/Launcher/Launcher.cpp]: we have/should use DisplayCluster version here?

File: apps/Launcher/Launcher.h [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+#ifndef LAUNCHER_H
+#define LAUNCHER_H
+
+#include <QGuiApplication>
+#include <QScopedPointer>
+
+class Launcher : public QGuiApplication
+{
+    Q_OBJECT
```
Comment (tribal-tec) [inline apps/Launcher/Launcher.h]: not needed imo

**Pull request closed without merging**